src/qtAliceVision/SequenceCache.cpp

```diff
@@ -199,6 +200,35 @@ QVariantList SequenceCache::getCachedFrames() const
     return intervals;
 }
 
+void SequenceCache::setFetchingSequence(bool fetching)
+{
+    _fetchingSequence = fetching;
+    abortPrefetching = !fetching;
+
+    if (fetching)
+    {
+        // Update internal state
+        _loading = true;
+
+        // Gather images to load
+        std::vector<FrameData> toLoad = _sequence;
+
+        // For now fill the allow worker thread to fill the whole cache capacity
+        const double fillRatio = 1.;
+
+        // Create new runnable and launch it in worker thread (managed by Qt thread pool)
+        auto ioRunnable = new PrefetchingIORunnable(_cache, toLoad, 0, fillRatio, _sequenceId.loadAcquire());
+        connect(ioRunnable, &PrefetchingIORunnable::progressed, this, &SequenceCache::onPrefetchingProgressed);
+        connect(ioRunnable, &PrefetchingIORunnable::done, this, &SequenceCache::onPrefetchingDone);
+        _threadPool.start(ioRunnable);
```

Comment:
Do not duplicate code, just move it into a dedicated function and use it in both places.